src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
-            for (Contact con : mapContactIdContactOwnerChange.values()) {
-            	setAccId.add(con.AccountId);
+    private void updateAccountPrimaryContact(Map<Id,Id> updateAccPrimaryContact, DmlWrapper dmlWrapper) {
+        List<Account> accountsToUpdate = new List<Account>();
+
+        for (Id i : updateAccPrimaryContact.keySet()) {
+            Account a = new Account(Id = i, Primary_Contact__c = updateAccPrimaryContact.get(i));
+            //Inject the Adv logic to handle npsp primary contact of the Account
+            if (Advancement_Info.useAdv() && Advancement_Info.getApiHEDA() != NULL) {
+                Advancement_Info.getApiHEDA().primaryContact(a, updateAccPrimaryContact.get(i));
             }
+            accountsToUpdate.add(a);
+        }
 
-            list<Account> listAcc = [Select Id, RecordTypeId, Primary_Contact__c, OwnerId, (select Id from Contacts limit 2) from Account
-                where Id IN :setAccId];
-
-            for (Account acc : listAcc) {
-                // only consider HH and Admin Accounts
-                //TODO: use setting to determine if Account owner should be changed (W-009780)
-                if (acc.RecordTypeId != null
-                        && (acc.RecordTypeId == userDefinedHHRecordTypeId
-                            || acc.RecordTypeId == userDefinedAdminRecordTypeId)) {
-                    // if there is only 1 contact associated with this account
-                    if (acc.Contacts.size() == 1 && acc.Primary_Contact__c != null) {
-                    	Contact c = mapContactIdContactOwnerChange.get(acc.Primary_Contact__c);
-                    	Id ownerIdContact = (c != null ? c.OwnerId : null);
-                    	if (acc.OwnerId != ownerIdContact) {
-                    		acc.OwnerId = ownerIdContact;
-                    		listAccToUpdate.add(acc);
-                    	}
-                    }
+        if (accountsToUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)accountsToUpdate);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Contact's owner has changed. Consider updating related account owner.
+    * @param contactOwnerChange a map of Contact Id to Contact, for the contacts to consider.
+    * @param dmlWrapper to hold the Contacts that need update.
+    */
+    private void updateOwners(Map<Id,Contact> contactOwnerChange, DmlWrapper dmlWrapper) {
+        List<Account> accountsToUpdate = new List<Account>();
+        Set<Id> contactsAccountId = new Set<Id>();
+
+        if (contactOwnerChange.size() == 0) {
+            return;
+        }
+
+        for (Contact con : contactOwnerChange.values()) {
+            contactsAccountId.add(con.AccountId);
+        }
+
+        List<Account> listAcc = [SELECT Id, RecordTypeId, Primary_Contact__c, OwnerId,
+                                (SELECT Id FROM Contacts limit 2)
```

Comment:
I may need a walkthrough a bit with the subqueries. What if we have 3 contacts that come back, we aren't always guaranteed to get them in a particular order, etc. (Also "limit" to LIMIT). It may become more clear to me as I walk through more, so pardon the questions with these.

Reply:
@lifewithryan, I'm actually not sure why there's limit to these subqueries. This was the original subqueries so I'm not sure if there's a use case/reason that we're missing.